src/payment/strategies/masterpass/masterpass-payment-strategy.ts

```diff
@@ -0,0 +1,153 @@
+import { PaymentStrategy } from '../';
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions
+} from '../../';
+import { Checkout, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType
+} from '../../../common/error/errors';
+import { bindDecorator as bind } from '../../../common/utility';
+import { StoreConfig } from '../../../config';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import PaymentMethod from '../../payment-method';
+
+import { MasterpassCheckoutCallback, MasterpassCheckoutOptions } from './masterpass';
+import MasterpassScriptLoader from './masterpass-script-loader';
+
+export default class MasterpassPaymentStrategy extends PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _walletButton?: HTMLElement;
+    private _walletCallback?: MasterpassCheckoutCallback;
+
+    constructor(
+        store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(methodId);
+
+        if (!this._paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!options.masterpass) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.masterpass" argument is not provided.');
+        }
+
+        const walletButton = options.masterpass.walletButton && document.getElementById(options.masterpass.walletButton);
```

Comment:
We should call `getElementById` after `_masterpassClientSetup`. Otherwise, we won't be able to select it if the element ID is dynamically rendered in the view.